core/utils/rename_dictionary_keys.py

```diff
@@ -1,14 +1,13 @@
-def rename_dictionary_keys(dictionary, corresp):
+def rename_dictionary_keys(list_dictionary, corresp):
     """
     Renomeia as chaves de um dicionário com base em um dicionário de correspondência.
 
     Args:
-        dictionary (dict): Dicionário.
+        list_dictionary (list): Lista de dicionário.
 
     Returns:
         dict: Um novo dicionário com as chaves atualizadas de acordo com o dicionário de correspondência.
     """
 
-    return {
-        corresp[key] if key in corresp else key: dictionary[key] for key in dictionary
-    }
+
+    return {corresp[key] if key in corresp else key: dictionary[key] for dictionary in list_dictionary for key in dictionary}
```

Comment:
@samuelveigarangel acho melhor colocar isso em bloco porque está pouco legível

Reply:
Ou aqui vc volta a receber apenas 1 dicionário e na chamada vc faz a iteração

Reply:
O que erro que estava acontecendo era que, em alguns casos, a função estava recebendo mais de um dicionário por lista. Então, como forma de pegar todos os dados, eu faço esse for adicional para percorrer todos os dict.